src/Json.Schema.ToDotNet/EqualityComparerGenerator.cs

```diff
@@ -699,11 +699,28 @@ private StatementSyntax GenerateObjectModelTypeHashCodeContribution(ExpressionSy
                             hashCodeContribution))));
         }
 
-        private StatementSyntax GenerateDictionaryHashCodeContribution(ExpressionSyntax expression)
+        private StatementSyntax GenerateDictionaryHashCodeContribution(string hashKindKey, ExpressionSyntax expression)
         {
             string xorValueVariableName = _localVariableNameGenerator.GetNextXorVariableName();
             string collectionElementVariableName = _localVariableNameGenerator.GetNextCollectionElementVariableName();
 
+            ExpressionSyntax keyExpression = SyntaxFactory.MemberAccessExpression(
```

Comment:
This is the only logic change to existing EqualityComparerGenerator. It's a fix for generating code for dictionary with data model type element.

Please see this PR ([https://github.com/microsoft/sarif-sdk/pull/2468](https://github.com/microsoft/sarif-sdk/pull/2468)) for more details